x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/process/AnalyticsProcessManager.java

```diff
@@ -438,12 +443,13 @@ private AnalyticsProcessConfig createProcessConfig(DataFrameDataExtractor dataEx
                                                            ExtractedFields extractedFields) {
             DataFrameDataExtractor.DataSummary dataSummary = dataExtractor.collectDataSummary();
             Set<String> categoricalFields = dataExtractor.getCategoricalFields(config.getAnalysis());
+            int threads = config.getMaxNumThreads() == null ? 1 : Math.min(config.getMaxNumThreads(), numAllocatedProcessors);
```

Comment:
I'm wondering where we should apply the default: here or in `DataFrameAnalyticsConfig`.
If the calling code doesn't care whether the user has set `max_num_threads`, the default of `1` could be as well applied in `DataFrameAnalyticsConfig`'s constructor.
Then we wouldn't need `null` handling here.
I don't have a strong opinion though so feel free to disregard the comment.

Reply:
I intentionally left it there as it's minimum effort. I anticipate we might change the way we implement this setting depending on feedback we'll get from testing.